epcpy/utils/parsers.py

```diff
@@ -170,6 +259,71 @@ def epc_pure_identity_to_tag_encodable(epc_pure_identity_uri: str) -> TagEncodab
     return scheme
 
 
+def gs1_element_string_to_gs1_element(
+    gs1_element_string: str, company_prefix_length: int
+) -> GS1Element:
+    """EPC gs1 element string to GS1Element class
+
+    Args:
+        gs1_element_string (str): GS1 element string
+
+    Raises:
+        ConvertException: Scheme is not GS1Element
+
+    Returns:
+        GS1Element: GS1Element class for this gs1 element string
+    """
+    for regex, scheme in GS1_ELEMENT_STRING_REGEX_TO_SCHEME.items():
+        if regex.fullmatch(gs1_element_string):
+            return scheme.from_gs1_element_string(
+                gs1_element_string, company_prefix_length
+            )
+
+    raise ConvertException(message=f"Unknown GS1 element string: {gs1_element_string}")
+
+
+MAX_ALLOWED_PATTERNS = {
```

Comment:
You want a set of strings and don't use the dict values?

Reply:
Would also prefer defining classes and subsequently something like this https://github.com/nedap/retail-epcpy/blob/a5f7ee14e97170448ba0c89cd24b1a9990cd4c2c/epcpy/utils/parsers.py#L111

Reply:
I'm not quite sure about this one, since the modelling of IDPAT for this case is strictly related to gs1 keys and an SGTIN class can never be "correctly" constructed from an IDPAT that contains * escapes. However, a gs1 key can be constructed if only 1 * is present, so we take a shortcut by constructing a temporary SGTIN with 0 as serial in order to retrieve its gs1 key. As such  I dont think its appropriate to apply these as class attributes.

Reply:
I do agree with removing the dictionary though

Reply:
I like your rewrite, that is what I meant.